quickstarts/microsoft.network/azure-bastion/main.bicep

```diff
@@ -0,0 +1,86 @@
+@description('Name of new or existing vnet to which Azure Bastion should be deployed')
+param vnetName string = 'vnet01'
+
+@description('IP prefix for available addresses in vnet address space')
+param vnetIpPrefix string = '10.1.0.0/16'
+
+@allowed([
+  'new'
+  'existing'
+])
+@description('Specify whether to provision new vnet or deploy to existing vnet')
+param vnetNewOrExisting string = 'new'
+
+@description('Bastion subnet IP prefix MUST be within vnet IP prefix address space')
+param bastionSubnetIpPrefix string = '10.1.1.0/27'
+
+@description('Name of Azure Bastion resource')
+param bastionHostName string
+
+@description('Azure region for Bastion and virtual network')
+param location string = resourceGroup().location
+
+var publicIpAddressName = '${bastionHostName}-pip'
+var bastionSubnetName = 'AzureBastionSubnet'
+
+resource publicIp 'Microsoft.Network/publicIpAddresses@2020-05-01' = {
+  name: publicIpAddressName
+  location: location
+  sku: {
+    name: 'Standard'
+  }
+  properties: {
+    publicIPAllocationMethod: 'Static'
+  }
+}
+
+resource virtualNetwork 'Microsoft.Network/virtualNetworks@2020-05-01' = if (vnetNewOrExisting == 'new') {
+  name: vnetName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        vnetIpPrefix
+      ]
+    }
+    subnets: [
+      {
+        name: bastionSubnetName
+        properties: {
+          addressPrefix: bastionSubnetIpPrefix
+        }
+      }
+    ]
+  }
+}
+
+resource subnet 'Microsoft.Network/virtualNetworks/subnets@2020-05-01' = if (vnetNewOrExisting == 'existing') {
+  parent: virtualNetwork
+  name: bastionSubnetName
+  properties: {
+    addressPrefix: bastionSubnetIpPrefix
+  }
+}
+
+resource bastionHost 'Microsoft.Network/bastionHosts@2020-05-01' = {
+  name: bastionHostName
+  location: location
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'IpConf'
+        properties: {
+          subnet: {
+            id: subnet.id
```

Comment:
this needs to account for the existing vnet scenario...

Reply:
Looks like it is:

resource virtualNetwork 'Microsoft.Network/virtualNetworks@2020-05-01' = if (vnetNewOrExisting == 'new') {

resource subnet 'Microsoft.Network/virtualNetworks/subnets@2020-05-01' = if (vnetNewOrExisting == 'existing') {

Is something missing or not right?

Reply:
oh, it's backwards.... when it's a new vnet, how does subnet.id resolve?  since "subnet" is a property and not a resource?